### PR TITLE
 Add Support for Tekton Pipeline v1 APIs 

### PR DESCRIPTION
**Pull request by khrm**

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind feature

**Review by khrm — COMMENTED** [review]
/hold

**Comment by vdemeester** [comment]
So I think `results` is in a similar case as chains, see https://github.com/tektoncd/chains/issues/665#issuecomment-1425642130. Today `results` will work with `tektoncd/pipeline` release >= 0.43, as it will get object through the `v1beta1` API (and conversion webhook do the work for us).

This PR would make `results` require a `tektoncd/pipeline` release that supports `v1` ; so it would not work with any release prior to v0.43. Seems fine to me, but we need to make sure we are ok with this, that starting from the next release of `tektoncd/results`, the minimum version of a `tektoncd/pipeline` release is going to be 0.43.

**Comment by adambkaplan** [comment]
Per WG discussion - we are going to hold this PR until v0.5.0 is released, then we can cut over to v1.

See https://github.com/tektoncd/results/issues/303#issuecomment-1434097748

**Comment by adambkaplan** [comment]
Keeping the hold on this PR - the logs feature is likely blocked by https://github.com/tektoncd/cli/issues/1837

**Comment by khrm** [comment]
@alan-ghelardi @dibyom @adambkaplan @sayan-biswas @enarha We should try to get this merge asap.

File: tools/tkn-results/go.mod
Comment (adambkaplan) [inline tools/tkn-results/go.mod:1]: It looks like we have a few obsolete dependencies as well. Maybe we take this up in separate pull requests?

File: tools/simpleui/main.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ type ui struct {
 }
 
 func main() {
-	conn, err := grpc.Dial("localhost:8080", grpc.WithInsecure(), grpc.WithBlock())
+	conn, err := grpc.Dial("localhost:50051", grpc.WithInsecure(), grpc.WithBlock())
```
Comment (adambkaplan) [inline tools/simpleui/main.go:71]: Why change the port number?

File: pkg/api/server/v1alpha2/record/record.go [hunk 2]
```diff
@@ -180,9 +181,9 @@ func validateData(m *pb.Any) error {
 	}
 	switch m.GetType() {
 	case "pipeline.tekton.dev/TaskRun":
-		return json.Unmarshal(m.GetValue(), &v1beta1.TaskRun{})
+		return json.Unmarshal(m.GetValue(), &v1.TaskRun{})
```
Comment (adambkaplan) [inline pkg/api/server/v1alpha2/record/record.go:184]: Will this break if the value data contains `v1beta1.TaskRun`?

File: pkg/api/server/v1alpha2/record/record.go [hunk 3]
```diff
@@ -180,9 +181,9 @@ func validateData(m *pb.Any) error {
...
 	case "pipeline.tekton.dev/PipelineRun":
-		return json.Unmarshal(m.GetValue(), &v1beta1.PipelineRun{})
+		return json.Unmarshal(m.GetValue(), &v1.PipelineRun{})
```
Comment (adambkaplan) [inline pkg/api/server/v1alpha2/record/record.go:186]: Will this break if the value data contains `v1beta1.PipelineRun`?

File: pkg/watcher/reconciler/pipelinerun/reconciler.go [hunk 4]
```diff
@@ -93,15 +93,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
...
+	pipelineRun, ok := object.(*pipelinev1.PipelineRun)
 	if !ok {
-		return false, fmt.Errorf("unexpected object (must not happen): want %T, but got %T", &pipelinev1beta1.PipelineRun{}, object)
+		return false, fmt.Errorf("unexpected object (must not happen): want %T, but got %T", &pipelinev1.PipelineRun{}, object)
 	}
 
 	logger := logging.FromContext(ctx)
 
-	// Support both minimal and full embedded status (see the TODO comment
-	// below).
```
Comment (adambkaplan) [inline pkg/watcher/reconciler/pipelinerun/reconciler.go:104]: +1

File: pkg/watcher/reconciler/pipelinerun/reconciler.go [hunk 5]
```diff
@@ -120,33 +118,12 @@ func (r *Reconciler) areAllUnderlyingTaskRunsReadyForDeletion(ctx context.Contex
 				return false, nil
 			}
 		}
-	} else {
-		// TODO(alan-ghelardi): remove this else once we support only
-		// Tekton v1 API since the full embedded status will no longer
-		// be supported.
```
Comment (adambkaplan) [inline pkg/watcher/reconciler/pipelinerun/reconciler.go:126]: +1 - great work @alan-ghelardi to mark this out ahead of time!

File: pkg/watcher/reconciler/reconciler_test.go
Comment (adambkaplan) [inline pkg/watcher/reconciler/reconciler_test.go:1]: Please remove this file. It was the source of a data race condition in our tests. We have replaced this "integration" controller test with a full end to end test.

File: tools/postgres-migrate/main.go
Comment (adambkaplan) [inline tools/postgres-migrate/main.go:1]: Changes to posgres-migrate are probably not necessary. This is a legacy tool from when this project used MySQL.

File: tools/postgres-migrate/main_test.go
Comment (adambkaplan) [inline tools/postgres-migrate/main_test.go:1]: Changes to postgres-migrate are probably not necessary.

File: tools/postgres-migrate/testdata_test.go
Comment (adambkaplan) [inline tools/postgres-migrate/testdata_test.go:1]: Changes to postgres-migrate are probably not necessary

**Review by adambkaplan — COMMENTED** [review]
Most of the code upgrades look good.

I do have a few questions regarding upgrade scenarios:

1. Can the apiserver handle updates from v1beta1 and v1 objects (ex - rolling upgrade of watchers)
2. Do we record the object version in the underlying record? Will clients be able to decode v1beta1 and v1 objects?

**Comment by adambkaplan** [comment]
/hold

At the last working group meeting, @enarha wanted to take the lead on publishing a new release this week. We should wait on merging this change until that is complete.

File: tools/simpleui/main.go (hunk 1 above)
Comment (khrm) [reply tools/simpleui/main.go:71]: I think this was changed in a different pr. So will update this. It happened due to rebasing.

**Comment by khrm** [comment]
We need to create an implementation based on WG call discussions and TEP in the community.

**Pull request closed without merging**